caddyhttp/httpserver/context.go

```diff
@@ -288,3 +288,33 @@ func (c Context) Map(values ...interface{}) (map[string]interface{}, error) {
 	}
 	return dict, nil
 }
+
+// Files reads and returns a slice of names from the given directory
+// relative to the root of Context c.
+func (c Context) Files(name string) ([]string, error) {
+	dir, err := c.Root.Open(path.Clean(name))
+	if err != nil {
+		return nil, err
+	}
+	defer dir.Close()
+
+	stat, err := dir.Stat()
+	if err != nil {
+		return nil, err
+	}
+	if !stat.IsDir() {
+		return nil, fmt.Errorf("%v is not a directory", name)
+	}
+
+	dirInfo, err := dir.Readdir(0)
+	if err != nil {
+		return nil, err
+	}
+
+	var names []string
```

Comment:
instead of append (and growing names) you can init it to len(dirInfo): 
```names := make([]string, len(dirInfo))``` and then:

```
for i, fileInfo := range dirInfo {
    names[i] = fileInfo.Name()
}
```

In this case you know exactly what is the size of ```names```